107/src/main/java/org/ehcache/jsr107/Eh107CacheStatisticsMXBean.java

```diff
@@ -204,42 +202,39 @@ private static float normalize(float value) {
     return (OperationStatistic<T>) result.iterator().next().getContext().attributes().get("this");
   }
 
-  <T extends Enum<T>> OperationStatistic<T> findAuthoritativeTierStatistic(Cache<?, ?> cache, Class<T> type, String statName) {
-    Query storeQuery = queryBuilder()
-        .children()
-        .children()
-        .filter(context(attributes(Matchers.<Map<String, Object>>allOf(
-            hasAttribute("tags", new Matcher<Set<String>>() {
-              @Override
-              protected boolean matchesSafely(Set<String> object) {
-                return object.containsAll(Collections.singleton("store"));
-              }
-            })))))
-        .build();
-
-    Set<TreeNode> storeResult = storeQuery.execute(Collections.singleton(ContextManager.nodeFor(cache)));
-    if (storeResult.size() > 1) {
-      throw new RuntimeException("store result must be unique");
-    }
-    if (storeResult.isEmpty()) {
-      throw new RuntimeException("store result must not be null");
-    }
-    Object authoritativeTier = storeResult.iterator().next().getContext().attributes().get("authoritativeTier");
+  <T extends Enum<T>> OperationStatistic<T> findLowestTierStatistic(Cache<?, ?> cache, Class<T> type, String statName) {
 
     Query statQuery = queryBuilder()
-        .children()
+        .descendants()
         .filter(context(attributes(Matchers.<Map<String, Object>>allOf(hasAttribute("name", statName), hasAttribute("type", type)))))
         .build();
 
-    Set<TreeNode> statResult = statQuery.execute(Collections.singleton(StatisticsManager.nodeFor(authoritativeTier)));
-    if (statResult.size() > 1) {
-      throw new RuntimeException("stat result must be unique");
+    Set<TreeNode> statResult = statQuery.execute(Collections.singleton(ContextManager.nodeFor(cache)));
+
+    if(statResult.size() < 1 || statResult.size() > 3) {
```

Comment:
Why the upper bound? I am not sure it brings us anything, as long as we have one eviction stat, we can find the proper one.
And not having the upper bound means less code to change in case we end up allowing more than 3 tiers.